mobile/src/people/people-room.tsx

```diff
@@ -25,13 +25,39 @@ const mapDispatchToProps = (dispatch: Dispatch<any>): EmployeesListDispatchProps
     onItemClicked: (employee: Employee) => dispatch( openEmployeeDetailsAction(employee))
 });
 
-export class PeopleRoomImpl extends React.Component<PeopleRoomProps & EmployeesListDispatchProps> {  
+export class PeopleRoomImpl extends React.Component<PeopleRoomProps & EmployeesListDispatchProps> {
+    public shouldComponentUpdate(nextProps: PeopleRoomProps & EmployeesListDispatchProps) {
+        if (this.props.onItemClicked !== nextProps.onItemClicked
+            || this.props.userEmployee !== nextProps.userEmployee
+        ) {
+            return true;
+        }
+
+        const predicate = (employee: Employee) => {
+            return this.props.userEmployee && employee.roomNumber === this.props.userEmployee.roomNumber;
+        };
+
+        const nextEmployees = nextProps.employees.employeesById.toArray().filter(predicate);
+
+        for (const employee of nextEmployees) {
```

Comment:
I think it won't work if you have an employee removed from the room.

Reply:
Try to replace it with filtering both collections first and then comparing them with https://facebook.github.io/immutable-js/docs/#/Map/equals

Reply:
👍

Reply:
works like a charm :)